src/grandorgue/sound/GOSoundFilter.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2006 Milan Digital Audio LLC
+ * Copyright 2009-2024 GrandOrgue contributors (see AUTHORS)
+ * License GPL-2.0 or later
+ * (https://www.gnu.org/licenses/old-licenses/gpl-2.0.html).
+ */
+
+#ifndef GOSOUNDFILTER_H_
+#define GOSOUNDFILTER_H_
+
+#include <cmath>
+
+enum class GO_FILTER_TYPE { LPF, HPF, LOW_SHELF, HIGH_SHELF };
+
+class GOSoundFilter {
+public:
+  class FilterState {
+  public:
+    FilterState() { Init(nullptr); }
+    void Init(const GOSoundFilter *filter) {
+      m_filter = filter;
+      for (int i = 0; i < 2; i++)
+        m_state[i] = 0;
+    }
+    bool IsToApply() { return m_filter->IsToApply(); }
+    void ProcessBuffer(unsigned n_blocks, float *buffer);
+
+  private:
+    float m_state[2];
+    const GOSoundFilter *m_filter;
```

Comment:
```suggestion
    const GOSoundFilter *p_filter;
```

Reply:
Usually it is difficult to remember who is an owner of the pointed object, whether it is necessary to delete it when the owning object is destroyed.

My suggested mnemonic is to use different prefixes: `m_` for owned objects and `p_` for not owned objects. So I asked you to change the name prefix from `m_` to `p_`.

Reply:
No real objection to it. I'll change the declaration and all the places where necessary, if it makes you happy.